terraform/datadog/locals.tf

```diff
@@ -25,11 +25,72 @@ EOF
 
   ## Contact details
   ## Both name and email are required for this block, so don't add if either are missing from platform-config.yml
-  contacts = var.config.contact_name == null || var.config.contact_email == null ? "" : <<EOF
-  contacts:
-    - name: ${var.config.contact_name}
+  ## This supports the 'old' way of setting email contact in platform-config.yml like this:
+  #contact_name: DBT Platform Engineers, Dave Glover
+  #contact_email: dbt-platform-engineers@digital.trade.gov.uk, david.glover@digital.trade.gov.uk
+  contact_name  = var.config.contact_name != null ? split(",", var.config.contact_name) : null
+  contact_email = var.config.contact_email != null ? split(",", var.config.contact_email) : null
+  contacts      = var.config.contact_name == null || var.config.contact_email == null ? "" : <<EOF
+  %{for k, v in local.contact_name}
+    - name: ${v}
       type: email
-      contact: ${var.config.contact_email}
+      contact: ${local.contact_email[k]}
+  %{endfor}
+EOF
+
+  ## This supports the 'new' way of setting contacts in platform-config.yml like this:
+  # contacts:
+  #   email:
+  #     - name: DBT Platform Engineers
+  #       address: dbt-platform-engineers@digital.trade.gov.uk
+  #     - name: Dave Glover
+  #       address: david.glover@digital.trade.gov.uk
+  #   slack:
+  #     - name: platform-squad-3
+  #       address: https://ditdigitalteam.slack.com/archives/C08FASZ72LS
+  #   link:
+  #     - name: DBT Platform Team Intranet Page
+  #       address: https://workspace.trade.gov.uk/teams/dbt-platform-team/?sub_view=people
+  contact_email_check = try(var.config.contacts.email, false)
+  contact_slack_check = try(var.config.contacts.slack, false)
+  contact_link_check  = try(var.config.contacts.link, false)
+  contact_teams_check = try(var.config.contacts.teams, false)
+  contacts_new        = <<EOF
+  %{if local.contact_email_check != false}
+    %{for k, v in var.config.contacts.email}
+      - name: ${v.name}
+        type: email
```

Comment:
Could we not simplify this down by using the same schema as DataDog? i.e. asking service teams to specify their contacts as...

```
- name: blah
   type: blah
   contact: blah
```

Reply:
Yes, no need to complicate things, and means that if DD supports a new contact type, we don't need to update our abstraction to support it.